File: app/src/main/java/com/wilinz/yuetingmusic/data/repository/MusicInfoRepository.java
package com.wilinz.yuetingmusic.data.repository;

import com.wilinz.yuetingmusic.data.AppNewWork;
import com.wilinz.yuetingmusic.data.model.MusicUrl;

import java.util.Arrays;
import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import kotlin.collections.CollectionsKt;
import retrofit2.http.Query;

public class MusicInfoRepository {

    public Observable<MusicUrl> getMusicUrls(List<Long> idList) {
        String query = String.join(",", CollectionsKt.map(idList, Object::toString));
        return AppNewWork.getInstance().musicInfoService.getMusicUrls(query);
    }

    private static volatile MusicInfoRepository singleton;

    private MusicInfoRepository() {
    }

    public static MusicInfoRepository getInstance() {
        if (singleton == null) {
            synchronized (MusicInfoRepository.class) {
                if (singleton == null) {
                    singleton = new MusicInfoRepository();
                }
            }
        }
        return singleton;
    }
}

File: app/src/main/java/com/wilinz/yuetingmusic/data/repository/MusicInfoRepository.kt
package com.wilinz.yuetingmusic.data.repository

import com.wilinz.yuetingmusic.data.AppNewWork
import com.wilinz.yuetingmusic.data.model.MusicUrl
import io.reactivex.rxjava3.core.Observable
import java.lang.String
import kotlin.Any
import kotlin.Long
import kotlin.synchronized

class MusicInfoRepository private constructor() {
    fun getMusicUrls(idList: List<Long>): Observable<MusicUrl> {
        val query = String.join(",", idList.map { obj: Any -> obj.toString() })
        return AppNewWork.instance!!.musicInfoService.getMusicUrls(query)
    }

    companion object {
        @Volatile
        private var singleton: MusicInfoRepository? = null
        val instance: MusicInfoRepository?
            get() {
                if (singleton == null) {
                    synchronized(MusicInfoRepository::class.java) {
                        if (singleton == null) {
                            singleton = MusicInfoRepository()
                        }
                    }
                }
                return singleton
            }
    }
}